pkg/utils/kubernetes/crd.go

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package kubernetes
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/exp/maps"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+	"github.com/gardener/gardener/pkg/utils/retry"
+)
+
+var (
+	// CRDWaitTimeout specifies the total time to wait for CRDs to become ready.
+	CRDWaitTimeout = 15 * time.Second
+)
+
+// CRDDeployer is a DeployWaiter that can deploy CRDs and wait for them to be ready.
+type CRDDeployer struct {
+	client            client.Client
+	applier           kubernetes.Applier
+	crdNameToManifest map[string]string
+}
+
+// NewCRDDeployer returns a DeployWaiter that can deploy CRDs and wait for them to be ready.
+func NewCRDDeployer(client client.Client, applier kubernetes.Applier, manifests []string) (*CRDDeployer, error) {
```

Comment:
```suggestion
func NewCRDDeployer(client client.Client, applier kubernetes.Applier, manifests []string) (component.DeployWaiter, error) {
```

Reply:
Thank you for the oversight :)
Because of the circular dependencies we talked about, I will move this to `components/crddeployer`